packages/website/src/components/editor/createProvideTwoslashInlay.ts

```diff
@@ -0,0 +1,100 @@
+// The following code is adapted from the code in microsoft/TypeScript-Website.
+// Source: https://github.com/microsoft/TypeScript-Website/blob/c8b2ea8c8fc216c163fe293650b2666c8563a67d/packages/playground/src/twoslashInlays.ts
+// License: https://github.com/microsoft/TypeScript-Website/blob/c8b2ea8c8fc216c163fe293650b2666c8563a67d/LICENSE-CODE
+
+import type Monaco from 'monaco-editor';
+import type * as ts from 'typescript';
+
+import type { SandboxInstance } from './useSandboxServices';
+
+const twoslashQueryRegex = /^(\s*\/\/\s*\^\?)\s*$/gm;
+function findTwoshashQueries(code: string): RegExpExecArray[] {
```

Comment:
[Refactor] This might be working now, but in general it's safer to recreate exec-ed regexes each time the execs are started.

```suggestion
function findTwoshashQueries(code: string): RegExpExecArray[] {
  const twoslashQueryRegex = /^(\s*\/\/\s*\^\?)\s*$/gm;
```

Reply:
You're right, good catch! About a year ago I spent _two days_ catching a bug in `vuejs/core`. It turned out to be caused by one small stateful regex with the global flag......